File: example/systems/HudRenderSystem.java
package net.brokenspork.systems;

import java.util.HashMap;

import net.brokenspork.components.Position;
import net.brokenspork.components.Sprite;
import net.brokenspork.core.Constants;

import com.artemis.ComponentMapper;
import com.artemis.annotations.Mapper;
import com.artemis.systems.VoidEntitySystem;
import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.graphics.OrthographicCamera;
import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.graphics.Texture.TextureFilter;
import com.badlogic.gdx.graphics.g2d.BitmapFont;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.badlogic.gdx.graphics.g2d.TextureAtlas;
import com.badlogic.gdx.graphics.g2d.TextureAtlas.AtlasRegion;
import com.badlogic.gdx.graphics.g2d.TextureRegion;

public class HudRenderSystem extends VoidEntitySystem {
	@Mapper
	ComponentMapper<Position> pm;
	@Mapper
	ComponentMapper<Sprite> sm;

	private HashMap<String, AtlasRegion> regions;
	private TextureAtlas textureAtlas;
	private SpriteBatch batch;
	private OrthographicCamera camera;
	private BitmapFont font;

	public HudRenderSystem(OrthographicCamera camera) {
		this.camera = camera;
	}

	@Override
	protected void initialize() {
		regions = new HashMap<String, AtlasRegion>();
		textureAtlas = new TextureAtlas("images-packed/pack.atlas");
		for (AtlasRegion r : textureAtlas.getRegions()) {
			regions.put(r.name, r);
		}

		batch = new SpriteBatch();

		Texture fontTexture = new Texture(Gdx.files.internal("fonts/normal_0.png"));
		fontTexture.setFilter(TextureFilter.Linear, TextureFilter.MipMapLinearLinear);
		TextureRegion fontRegion = new TextureRegion(fontTexture);
		font = new BitmapFont(Gdx.files.internal("fonts/normal.fnt"), fontRegion, false);
		font.setUseIntegerPositions(false);
	}

	@Override
	protected void begin() {
		batch.setProjectionMatrix(camera.combined);
		batch.begin();
	}

	@Override
	protected void processSystem() {
		batch.setColor(1, 1, 1, 1);
		font.draw(batch, "FPS: " + Gdx.graphics.getFramesPerSecond(), -(Constants.FRAME_WIDTH / 2) + 20, Constants.FRAME_HEIGHT / 2 - 20);
		font.draw(batch, "Active entities: " + world.getEntityManager().getActiveEntityCount(), -(Constants.FRAME_WIDTH / 2) + 20, Constants.FRAME_HEIGHT / 2 - 40);
		font.draw(batch, "Total created: " + world.getEntityManager().getTotalCreated(), -(Constants.FRAME_WIDTH / 2) + 20, Constants.FRAME_HEIGHT / 2 - 60);
		font.draw(batch, "Total deleted: " + world.getEntityManager().getTotalDeleted(), -(Constants.FRAME_WIDTH / 2) + 20, Constants.FRAME_HEIGHT / 2 - 80);
	}
	
	@Override
	protected void end() {
		batch.end();
	}

}

File: example/src/systems/HudRenderSystem.ts
module example.systems {

  import Container = PIXI.Container;
  import BitmapText = PIXI.extras.BitmapText;
  import EntitySystem = artemis.EntitySystem;
  import VoidEntitySystem = artemis.systems.VoidEntitySystem;
  import PositionComponent = artemis.components.PositionComponent;
  import SpriteComponent = artemis.components.SpriteComponent;


  export class HudRenderSystem extends VoidEntitySystem {
    private activeEntities:BitmapText;
    private totalCreated:BitmapText;
    private totalDeleted:BitmapText;

    public initialize() {
      //BitmapText
      //
      var font = {font: '36px Radio Stars', align: 'left'};
      this.activeEntities = new BitmapText("Active entities:           ", font);
      this.totalCreated = new BitmapText("Total created:          ", font);
      this.totalDeleted = new BitmapText("Total deleted:          ", font);

      this.activeEntities.scale.set(0.5);
      this.totalCreated.scale.set(0.5);
      this.totalDeleted.scale.set(0.5);

      this.activeEntities.position.set(0, 20);
      this.totalCreated.position.set(0, 40);
      this.totalDeleted.position.set(0, 60);

      bosco.viewContainer.addChild(this.activeEntities);
      bosco.viewContainer.addChild(this.totalCreated);
      bosco.viewContainer.addChild(this.totalDeleted);
    }


    public processSystem() {

      this.activeEntities.text = "Active entities: " + this.world.getEntityManager().getActiveEntityCount();
      this.totalCreated.text = "Total created: " + this.world.getEntityManager().getTotalCreated();
      this.totalDeleted.text = "Total deleted: " + this.world.getEntityManager().getTotalDeleted();

    }
  }
}
